test/src/kbasesearchengine/test/search/ElasticIndexingStorageTest.java

```diff
@@ -1815,4 +1815,45 @@ public void testDeleteObjectWithSubobject() throws Exception {
 
         assertThat("objects not deleted", res2, is(set()));
     }
+    @Test
+    public void testLastVersionIndexForObjectsWithSubobjects() throws Exception {
+        // regular object
+        indexObject("WS:2000/1/2", "WS:2000/1/2", 1);
+        indexObject("WS:2000/1/2", "WS:2000/1/2:sub/yay", 1);
+
+        indexStorage.refreshIndex("*");
+
+        indexObject("WS:2000/1/1", "WS:2000/1/1", 2);
+        indexObject("WS:2000/1/1", "WS:2000/1/1:sub/yay", 2);
+
+
+        //check that both objects are found
+        final Set<GUID> res = indexStorage.searchIds(
+                Collections.emptyList(),
+                MatchFilter.getBuilder().withNullableFullTextInAll("imaprettypony").build(),
+                null,
+                AccessFilter.create().withAccessGroups(2000));
+        assertThat("incorrect objects found", res,
+                is(set(new GUID("WS:2000/1/2"), new GUID("WS:2000/1/2:sub/yay"))));
+
+    }
+
+    public void indexObject(final String pguid, final String guid, final int ver)throws Exception{
```

Comment:
Should make it clear that the ver here is the search type version, not the source object version. That confused me for a bit.